services/dashboard/views/apps/apps.js

```diff
@@ -34,16 +34,25 @@ angular.module('dashboard')
         show: false
       });
 
-      $scope.openSubmitDialog = function() {
+      $scope.openSubmitGearAppDialog = function() {
+        submitWindow.$scope.isStormApp = false;
+        submitWindow.$promise.then(submitWindow.show);
+      };
+
+      $scope.openSubmitStormAppDialog = function() {
+        submitWindow.$scope.isStormApp = true;
         submitWindow.$promise.then(submitWindow.show);
       };
 
       $scope.composeMenuOptions = [{
+        text: '<i class="glyphicon glyphicon-none"></i> <b>Submit User Application</b>',
```

Comment:
There will be three options. Compose DAG is there. 

I think "Submit *Foo* Application" will be better. JAR is actually a part of an application. 
Regarding "Submit Gearpump Application" vs "Submit User Application", I think Gearpump is better than user, as we might submit a Gearpump built-in example manually, which is not an user self developed application.